### ISSUE TITLE
Add time/effort indication to templates

### ISSUE THREAD

**Issue by camerons** [thegooddocsproject/templates#63]
from: template-tutorial
Deepti Korwar
18:47 15 Nov
Consider adding the approximate time the user might take to complete this activity.

**Comment by camerons** [comment]
This is a topic which I think needs some deep thinking and development of a collective position.
I've heard two sides to sides to the mention of "indication of time".
Pro: 
* Adding "reading time" or "time to complete an exercise" gives the reader the opportunity of gauging beforehand how much time to set aside or whether it is worth investing the time in the activity.
Con:
* Time-to-complete can vary a lot depending on the expertise of the reader. Indicating a short time can result in readers feeling stupid, or worse, managers applying punishments for slow learners. Indicating a long time can mean that the reader won't even start.

**Comment by cwcromwell** [comment]
We can't estimate the time to complete any of these, because it depends on several variables that we can't control: 

*The complexity of the task documented
* The length of the document being created. (How many steps are in the tutorial?)
* The skill of the writer creating the tutorial. 

The time required to document two different tasks -- using the same template -- might be minutes (how to sign up for our WTD event) versus days (deploy our product into your Kubernetes cluster).

Using a template should reduce the level of time and effort required, regardless of the type of task you want to document. But that's different from knowing the amount of time required before you begin.

**Comment by camerons** [comment]
@cwcromwell The recommendation (as I interpret it), is that the end document writer should estimate the time taken to execute the quickstart or tutorial.

Us writers of the templates should provide end document writers guidance on how to come up with that estimate.
We probably should also suggest how long a quickstart/tutorial should take. (If it is too long, then suggest breaking into multiple tutorials.)

**Comment by bjnath** [comment]
I'd propose the word "typically," which links to a section explaining what we mean.

People look at the size and contents of the template and get their own idea of what's involved. What we should be doing is alerting them to gotchas -- this is not as easy as it looks, because..., or this is easier than it looks, because...

**Comment by camerons** [comment]
Notes from meeting discussion about this:

Aidan (from prior to meeting): I think this is the oldest issue, let’s discuss:
My thoughts: I would be reluctant to add a time that it takes to complete a tutorial to the tutorial template. In my experience tutorials are wildly variable in how long they take someone to compete, and I wouldn't encourage tech writers to include this. It may make some sense to add how long it takes to read an article or a page of documentation, so I might support that. But even that is debatable. My gut feeling is to let this go or initiate a broader conversation about it on Slack.
* Ryan: For longer tutorial helps to provide indicators.
* Morgan: (From working at a learning company. Add an estimated time, to schedule work)
* Need to work out reading time for tech docs. Consider images.
* Chunk concepts and provide signposting (e.g. page 4 of 7).
* We should err toward recording the longest time. (Better to make people feel good about reading fast than feel bad about reading slow.)
* Is there research out there on attention span?
* What is too long for a tutorial and then break it into smaller tutorials
* Morgan, in training company, always were breaking down long dense content into smaller concise.

**Comment by Loquacity** [comment]
I think we are too many steps removed to be able to provide guidance here. The person writing the actual document might be able to provide an estimate for how long it would take readers to read and/or perform actions described in the document, but I don't think there is enough evidence either for or against doing so that we could safely recommend that they a) provide an estimate, or b) tell them how to calculate that estimate. If we want to provide an estimate for how long it takes to write a document based on a template, I think it's too highly variable (for all the reasons @cwcromwell mentioned), which makes any estimate we give almost entirely useless: "between 10 minutes and three days" is not valuable. As for adding this to the base template, it's another step removed again, so I'm not inclined to include it. I think there are more important factors that writers should consider when creating new templates (like audience, voice, etc), and we don't want to be writing a university-level degree, so we should be somewhat judicious about what we include. 

TL;DR: I can't see value in including this at any level, and vote for closing this issue without action.

**Comment by aidanatwork** [comment]
+1 on closing without action

**Comment by viraji** [comment]
+1 on closing without action, for indicating a "reading time" on the
tutorial. That poses no real value as has been pointed out above.

Alternatively, however, we can *optionally* include some indication that
would gauge the level of competency required for the tutorial to be carried
out, as below for example:
Level of competency in handling API documentation for carrying out this
tutorial is : {low, medium, high}.
That's just a suggestion, and is likely to address the concerns that Ryan
and Morgan have raised above, because it still gives an indication of the
complexity involved.

/Viraji.

**Comment by camerons** [comment]
I'm +0 (mildly in favor) on closing this issue as it stands because I think if it were to apply anywhere, it should start by being applied to the base template.

I'm -0 (mildly against) dropping "expected reading time" from our template guidance. I think there is value is helping readers know how long a document will take to run/read. However, we can take that up as a separate issue later.

**Comment by aidanatwork** [comment]
Regarding competency, the most common convention I’ve seen is to say something like; “Basic understanding of HTML and CSS is needed for this tutorial.” Or “Basic understanding of Python is helpful for this tutorial, but not required.” Rather than saying low/medium/high, you address the specific knowledge that is needed as a prerequisite.

**Comment by jaredmorgs** [comment]
"Reading Time" is often an automatically calculated value based on word
count offered by some sites like Medium.

I feel that manually calculating this value is "finger in the air" stuff
best left out of our template recommendations.

I'm therefore +1 to closing of this issue as "won't do".

**Comment by camerons** [comment]
This was discussed in weekly meeting:
* Morgan: This is something that the toolchain can help with.
    * Want to use intelligence around types of content, such as code blocks, and the type of language.
* Alyssa: Eg: https://www.docebo.com/knowledge-base/elearning-how-to-manage-docebo-app-custom-domain/
* Cameron: If we include this, it is something that we should incorporate into the metadata in the base template, which we can track in the current proposal. https://github.com/thegooddocsproject/incubator/blob/camerons-base-template/base-template/base-template.md

Based on this, and prior comments, I'm closing this task with no action.
